includes/payment-methods/class-wc-stripe-express-checkout-helper.php

```diff
@@ -683,19 +683,28 @@ public function should_show_express_checkout_button() {
 		}
 
 		// Don't show on checkout if disabled.
-		if ( is_checkout() && ! $this->should_show_ece_on_checkout_page() ) {
+		// One Page Checkout plugin creates checkout functionality on product pages, so we need to check for it.
+		if ( ( is_checkout() || $this->is_one_page_checkout() ) && ! $this->should_show_ece_on_checkout_page() ) {
```

Comment:
Non-blocking nit: rather than calling `$this->is_one_page_checkout()` multiple times, I think it would be simpler to stash the value in a local variable, much like we're doing for `$is_product` below.
```suggestion
		$is_one_page_checkout = $this->is_one_page_checkout();

		// One Page Checkout plugin creates checkout functionality on product pages, so we need to check for it.
		if ( ( is_checkout() || $is_one_page_checkout ) && ! $this->should_show_ece_on_checkout_page() ) {
```

Reply:
Good idea! Updated in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4757/commits/11a0cc0e93d02b541a20581a203a0ea761b6dacb